src/write.c

```diff
@@ -3017,6 +3006,47 @@ static avifResult avifRWStreamWriteProperties(avifItemPropertyDedup * const dedu
             // We don't add an 'lsel' property since many decoders do not support it and will reject the image,
             // see https://github.com/AOMediaCodec/libavif/pull/2429
         }
+
+        // Also write the transformative properties.
+
+        if (item->itemCategory == AVIF_ITEM_COLOR) {
+            // Color specific properties
+            // Note the 'tmap' (tone mapped image) item when a gain map is present also has itemCategory AVIF_ITEM_COLOR.
+            AVIF_CHECKRES(avifEncoderWriteClapIrotImir(&dedup->s, s, itemMetadata, &item->ipma, dedup));
```

Comment:
Should we move line 2960:

```
    AVIF_CHECKRES(avifEncoderWriteHDRProperties(&dedup->s, s, itemMetadata, &item->ipma, dedup));
```

so that we always call `avifEncoderWriteColorProperties`, `avifEncoderWriteHDRProperties`, `avifEncoderWriteHDRProperties` in the same order in the `item->itemCategory == AVIF_ITEM_COLOR` case. See lines 3620-2624.